nemoguardrails/actions/llm/generation.py

```diff
@@ -1126,10 +1139,16 @@ async def generate_intent_steps_message(
                 _streaming_handler = StreamingHandler()
                 local_streaming_handlers[_streaming_handler.uid] = _streaming_handler
 
+                generation_options: GenerationOptions = generation_options_var.get()
+                additional_params = {
+                    **((generation_options and generation_options.llm_params) or {}),
+                    "temperature": self.config.lowest_temperature,
+                }
+
```

Comment:
Here it's ok to use the `llm_params` because everything is happening in single call.

Reply:
O a second thought, this is tricky too. For example, if the original prompt has a `max_tokens` set to 1, let's say we want to LLM to respond with yes/no, then this would break the logic as the single LLM call mode needs to alter the prompt and generate more tokens. I don't have a solution for this.